pkg/provider/cloud/vsphere/provider.go

```diff
@@ -48,109 +39,44 @@ const (
 	defaultCategoryPrefix = "kubermatic.k8c.io"
 )
 
-// Provider represents the vsphere provider.
-type Provider struct {
+// VSphere represents the vsphere provider.
+type VSphere struct {
 	dc                *kubermaticv1.DatacenterSpecVSphere
+	log               *zap.SugaredLogger
 	secretKeySelector provider.SecretKeySelectorValueFunc
 	caBundle          *x509.CertPool
 }
 
-// Folder represents a vsphere folder.
-type Folder struct {
-	Path string
-}
-
 // NewCloudProvider creates a new vSphere provider.
-func NewCloudProvider(dc *kubermaticv1.Datacenter, secretKeyGetter provider.SecretKeySelectorValueFunc, caBundle *x509.CertPool) (*Provider, error) {
+func NewCloudProvider(dc *kubermaticv1.Datacenter, secretKeyGetter provider.SecretKeySelectorValueFunc, caBundle *x509.CertPool) (*VSphere, error) {
 	if dc.Spec.VSphere == nil {
 		return nil, errors.New("datacenter is not a vSphere datacenter")
 	}
-	return &Provider{
+	return &VSphere{
 		dc:                dc.Spec.VSphere,
+		log:               log.Logger,
 		secretKeySelector: secretKeyGetter,
 		caBundle:          caBundle,
 	}, nil
 }
 
-var _ provider.CloudProvider = &Provider{}
+var _ provider.ReconcilingCloudProvider = &VSphere{}
 
-type Session struct {
-	Client     *govmomi.Client
-	Finder     *find.Finder
-	Datacenter *object.Datacenter
+func (v *VSphere) ReconcileCluster(ctx context.Context, cluster *kubermaticv1.Cluster, update provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
+	return v.reconcileCluster(ctx, cluster, update, true, true)
 }
 
-// Logout closes the idling vCenter connections.
-func (s *Session) Logout(ctx context.Context) {
-	if err := s.Client.Logout(ctx); err != nil {
-		kruntime.HandleError(fmt.Errorf("vSphere client failed to logout: %w", err))
-	}
-}
+func (v *VSphere) reconcileCluster(ctx context.Context, cluster *kubermaticv1.Cluster, update provider.ClusterUpdater, force bool, _ bool) (*kubermaticv1.Cluster, error) {
```

Comment:
If you don't need the second bool (`_ bool`), why not just drop it from the function signature?

Reply:
That's actually true. Will wait if the verify job is happy then will push the new PR adjustment